src/main/java/strategy/CalculateStrategy.java

```diff
@@ -0,0 +1,35 @@
+package strategy;
+
+import dao.Storage;
+import java.util.EnumMap;
+import java.util.Map;
+import model.FruitTransaction;
+import model.Operation;
+
+public class CalculateStrategy implements Strategy {
+    private final Storage storage;
+    private Map<Operation, FruitOperation> operationMap;
+
+    public CalculateStrategy(Storage storage) {
+        this.storage = storage;
+        this.operationMap = initializeOperationMap(storage);
+    }
```

Comment:
The code should not be implementation dependent
```suggestion
    public CalculateStrategy(Storage storage, Map<Operation, FruitOperation> operationMap) {
        this.storage = storage;
        this.operationMap = operationMap;
    }
```